pkg/operator/ceph/object/user/controller.go

```diff
@@ -278,41 +279,117 @@ func (r *ReconcileObjectStoreUser) reconcile(request reconcile.Request) (reconci
 	return reconcile.Result{}, *cephObjectStoreUser, nil
 }
 
-func (r *ReconcileObjectStoreUser) reconcileCephUser(cephObjectStoreUser *cephv1.CephObjectStoreUser) (reconcile.Result, error) {
-	err := r.createOrUpdateCephUser(cephObjectStoreUser)
+func (r *ReconcileObjectStoreUser) reconcileCephUser(cephObjectStoreUser *cephv1.CephObjectStoreUser) (reconcile.Result, bool, error) {
+	userKeysSource, err := r.createOrUpdateCephUser(cephObjectStoreUser)
 	if err != nil {
-		return reconcile.Result{}, errors.Wrapf(err, "failed to create/update object store user %q", cephObjectStoreUser.Name)
+		return reconcile.Result{}, userKeysSource, errors.Wrapf(err, "failed to create/update object store user %q", cephObjectStoreUser.Name)
 	}
 
-	return reconcile.Result{}, nil
+	return reconcile.Result{}, userKeysSource, nil
 }
 
-func (r *ReconcileObjectStoreUser) createOrUpdateCephUser(u *cephv1.CephObjectStoreUser) error {
+func (r *ReconcileObjectStoreUser) getUserDefinedSecret(u *cephv1.CephObjectStoreUser, userKeysSource bool) ([]admin.UserKeySpec, bool, error) {
+	var credentialsStruct []admin.UserKeySpec
+	secretName := object.GenerateCephUserSecretName(u.Spec.Store, u.Name)
+	namspacedName := types.NamespacedName{Namespace: u.Namespace, Name: secretName}
+
+	cephObjectStoreUserSecret := &corev1.Secret{}
+	err := r.client.Get(r.clusterInfo.Context, namspacedName, cephObjectStoreUserSecret)
+	if err != nil {
+		if !kerrors.IsNotFound(err) {
+			return credentialsStruct, userKeysSource, errors.Wrapf(err, "failed to get user cephobjectstoreuser secret %q", secretName)
+		}
+		logger.Debugf("no user secret %q provided for cephobjectstoreuser", secretName)
+
+	} else {
+		if secret.ForceUpdateObjectUserSecret(cephObjectStoreUserSecret.GetAnnotations()) {
+			userKeysSource = true
+			credentialsStruct, err = secret.UnmarshalKeys(cephObjectStoreUserSecret.Data["Credentials"])
+			if err != nil {
+				return credentialsStruct, userKeysSource, errors.Wrapf(err, "invalid cephobjectstoreuser secret %q format", secretName)
+			}
+			credentialsStruct, err = secret.ValidateCredentials(string(cephObjectStoreUserSecret.Data["AccessKey"]), string(cephObjectStoreUserSecret.Data["SecretKey"]), credentialsStruct)
+			if err != nil {
+				return credentialsStruct, userKeysSource, errors.Wrapf(err, "invalid cephobjectstoreuser secret %q values", secretName)
+			}
+		}
+	}
+
+	return credentialsStruct, userKeysSource, nil
+}
+
+func (r *ReconcileObjectStoreUser) modifyObjectUserSecret(credentialsStruct []admin.UserKeySpec, user admin.User) error {
+	// if secret exists and source of truth is secret then use the user specified keys
+	r.userConfig.Keys = credentialsStruct
+	// update the userid on the credentials
+	credentialsStruct = secret.UpdatecredentialsUserId(user.ID, user.DisplayName, credentialsStruct)
```

Comment:
When/why does Rook need to do this?

Reply:
As every value has a predefined userid, so to query a access key we need it